### ISSUE TITLE
[Bug]: destructuring-assignment rule false pass

### ISSUE THREAD

**Issue by grant-unwin-cc** [yannickcr/eslint-plugin-react#3561]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

I have identified a false pass for the destructuring assignment rule.

![image](https://user-images.githubusercontent.com/127852756/232733123-5a622480-4b6b-412d-9f3c-38a8cac112bb.png)

Note the, destructureInSignature set to 'always'.

I have made a branch with the test case added, which fails here:

https://github.com/grant-unwin-cc/eslint-plugin-react/blob/destructuring-assignment-false-pass-example/tests/lib/rules/destructuring-assignment.js#L895

### Expected Behavior

Expect the test to pass, it does not. It seems to be mislead by a destructuring further down.

https://github.com/grant-unwin-cc/eslint-plugin-react/blob/destructuring-assignment-false-pass-example/tests/lib/rules/destructuring-assignment.js#L895

### eslint-plugin-react version

master

### eslint version

8.38.0

### node version

18.15.0

**Comment by ljharb** [comment]
Good catch - can you make a PR, even if it just has the failing test cases?
